utils/python/CIME/XML/entry_id.py

```diff
@@ -157,12 +157,18 @@ def _set_value(self, node, value, vid=None, subgroup=None, ignore_type=False):
         subgroup is ignored in the general routine and applied in specific methods
         """
         expect(subgroup is None, "Subgroup not supported")
+        valid_values = self.get_optional_node("valid_values", root=node)
         if ignore_type:
             expect(type(value) is str, "Value must be type string if ignore_type is true")
-            node.set("value",value)
+            str_value = value
         else:
             type_str = self._get_type_info(node)
-            node.set("value", convert_to_string(value, type_str, vid))
+            str_value = convert_to_string(value, type_str, vid)
+        if valid_values is not None and valid_values.text is not None and not str_value.startswith('$'):
```

Comment:
Things look good and the check functionality does work. However, I have a question about our policy for resolving values. If I do something like `./xmlchange REST_OPTION=\$foo`, I get a warning in `./case.setup`:
> Not able to fully resolve item '$foo'

However, if `foo` has a valid value, `./case.setup` completes (as does `./case.build`).
What is our policy on when and whether we use environment values when resolving XML entries? Should foo be a valid XML key? If so, you should add a check for this above. If not, when does that environment variable have to be valid? Can we document the expected behavior of setting an XML variable to a variable-to-be-resolved-later?